doc/usecases/core/UC_keyset_index_access_many.md

```diff
@@ -0,0 +1,26 @@
+# Use Case: Index access to `KeySet`
+
+## Summary
+
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Caller accesses contents of `KeySet` by index
+
+## Scenarios
+
+- **Precondition:**
+  - [`KeySet` has been created](UC_keyset_create.md).
+  - [`Key`(s) have been inserted into `KeySet`](UC_keyset_insert.md).
+- **Main success scenario:**
+  - Caller requests `Key`s in valid index range (`[f, l)` with `0 <= f < size && f <= l <= size`) from `KeySet` KS
+  - Core creates new `KeySet` KS1
+  - Core adds all `Key`s in KS with index `i >= f && i < l` to KS1.
+  - Core returns KS1
+- **Alternative scenario:** -
+- **Error scenario:**
+  - Caller requests `Key` at invalid index range (`f < 0 || f >= size || l < f || l > size`) from `KeySet`
+  - Core returns `NULL`
+- **Postcondition:**
+  - The initial `KeySet` KS MUST be unmodified.
+- **Non-functional Constraints:** -
```

Comment:
Good question. I'd say it should be `O(N)` where `N` is the initial number of keys in KS. Which would be the same for removing a single key, because in both cases, we at minimum need to `memmove` a part of `ks->array` in the `KeySet`.

That's actually (partly) the reason for introducing this API, because  currently you need to "remove single key" M-times to remove M keys, which give `O(M*N)`.

Reply:
Then please add this info. I am wondering why someone even wants to request keys in an index range (as core feature).

Reply:
As stated in #4805 whether the index range functions are in `libelektra-core` or `libelektra-extra` is debatable. However, there are two clear use cases (which I will add to the file):
1. Getting all the `Key`s in a key hierarchy, i.e. what `ksCut`/`ksBelow` does right now. We need this index access function, because the key hierarchy is only returned as an index range (for consistency with lookup and flexibility).
2. Array manipulation. If you have an array `foo/#` and want to get, move or delete a section (e.g. `#4` until `#7`), then the index range stuff allows doing this. With just the hierarchy based APIs (like `ksCut`) and array is always one atomic block.

Reply:
> which I will add to the file

I didn't find them.